src/processor/blockchain-events/offer.ts

```diff
@@ -36,9 +36,12 @@ const handlers: HandlersObject<BlockchainOfferEvents, BlockchainEventProcessorOp
     const flag = firstMsg.substring(2, 4)
 
     if (flag === '01') { // PeerId definition
-      store.peerId = decodeByteArray([`0x${firstMsg.substring(4)}`, ...restMsg])
+      const offersPeerId = decodeByteArray([`0x${firstMsg.substring(4)}`, ...restMsg])
 
-      logger.info(`PeerId ${store.peerId} defined`)
+      // We don' update PeerId as it is already stored locally, but will Error if they don't match
+      if (store.peerId !== offersPeerId) {
```

Comment:
Should we mark somehow agreements with inconsistent `peerId`

Reply:
Well, I don't think there is much we can do here except let the user know that it is a mismatch. 

An alternative is to exit the daemon, but that solution seems to me bit too drastic.